meteor/server/publications/timeline.ts

```diff
@@ -50,6 +61,24 @@ meteorCustomPublishArray<RoutedTimeline>(
 		}
 	}
 )
+meteorCustomPublishArray<TimelineDatastoreEntry>(
+	PubSub.timelineDatastoreForDevice,
+	'studioTimelineDatastore',
+	function (pub, deviceId: PeripheralDeviceId, token) {
+		if (
+			PeripheralDeviceReadAccess.peripheralDeviceContent({ deviceId: deviceId }, { userId: this.userId, token })
+		) {
+			const peripheralDevice = PeripheralDevices.findOne(deviceId)
+
+			if (!peripheralDevice) throw new Meteor.Error('PeripheralDevice "' + deviceId + '" not found')
+
+			const studioId = peripheralDevice.studioId
+			if (!studioId) return []
+
+			createObserverForTimelineDatastorePublication(pub, PubSub.timelineDatastoreForDevice, studioId)
```

Comment:
From what I see, there's no transformation of data or caching going in the custom publish function. If that's the case we should not use the `meteorCustomPublishArray` since that adds unnecessary overhead (performance penalty) for the publication, rather we should just do a `meteorPublish`:
```typescript
meteorPublish(PubSub.timelineDatastoreForDevice, function (deviceId: string, token: string) {
	if (PeripheralDeviceReadAccess.peripheralDeviceContent({ deviceId: deviceId }, { userId: this.userId, token })) {
			const peripheralDevice = PeripheralDevices.findOne(deviceId)

			if (!peripheralDevice) throw new Meteor.Error('PeripheralDevice "' + deviceId + '" not found')

			const studioId = peripheralDevice.studioId
			if (!studioId) return null
			const modifier: FindOptions<TimelineDatastoreEntry> = {
				fields: {},
			}

			return TimelineDatastore.find({studioId}, modifier)
	}
	return null
})
```